dask/dataframe/io/parquet/arrow.py

```diff
@@ -1047,3 +1047,29 @@ def write_metadata(parts, fmd, fs, path, append=False, **kwargs):
                 _append_row_groups(_meta, parts[i][0]["meta"])
             with fs.open(metadata_path, "wb") as fil:
                 _meta.write_metadata_file(fil)
+
+    @classmethod
+    def collect_file_metadata(cls, path, fs, file_path):
+        with fs.open(path, "rb") as f:
+            meta = pq.ParquetFile(f).metadata
+        if file_path:
+            meta.set_file_path(file_path)
+        return meta
+
+    @classmethod
+    def aggregate_metadata(cls, meta_list, fs, out_path):
+        meta = None
+        for _meta in meta_list:
+            if meta:
+                _append_row_groups(meta, _meta)
+            else:
+                meta = _meta
```

Comment:
I was wondering if this could use `reduce` but I guess since the object is modified inplace that doesn't really work?

Reply:
Hmm - Not sure if `reduce` requires it to be a pure function (I completely forgot that function existed btw)

Reply:
It might work if you just made the function return the modified object.